src/components/ripple/ripple.ts

```diff
@@ -0,0 +1,173 @@
+   * calling start() and end().
+   */
+  @Input('disabled') disabled: boolean;
+  /**
+   * If set, the radius in pixels of foreground ripples when fully expanded. If unset, the radius
+   * will be the distance from the center of the ripple to the furthest corner of the element's
+   * bounding rectangle.
+   */
+  @Input('max-radius') maxRadius: number = 0;
+  /**
+   * If set, the normal duration of ripple animations is divided by this value. For example,
+   * setting it to 0.5 will cause the animations to take twice as long.
+   */
+  @Input('speedFactor') speedFactor: number = 1;
+  /** Custom color for ripples. */
+  @Input('color') color: string;
+  /** Custom color for the ripple background. */
+  @Input('backgroundColor') backgroundColor: string;
+
+  /** Whether the ripple background will be highlighted to indicated a focused state. */
+  @HostBinding('class.md-ripple-focused') @Input('focused') focused: boolean;
+  /** Whether foreground ripples should be visible outside the component's bounds. */
+  @HostBinding('class.md-ripple-unbounded') @Input('unbounded') unbounded: boolean;
+
+  private _rippleManager: MdInkRippleManager;
+
+  constructor(_elementRef: ElementRef) {
+    // These event handlers are attached to the element that triggers the ripple animations.
+    const eventHandlers = new Map<string, (e: Event) => void>();
+    eventHandlers.set('mousedown', (event: MouseEvent) => this._mouseDown(event));
+    eventHandlers.set('click', (event: MouseEvent) => this._click(event));
+    eventHandlers.set('mouseleave', (event: MouseEvent) => this._mouseLeave(event));
+    this._rippleManager = new MdInkRippleManager(_elementRef, eventHandlers);
+  }
+
+  /** TODO: internal */
+  ngOnInit() {
+    // If no trigger element was explicity set, use our parent.
+    if (!this.trigger) {
+      this._rippleManager.setTriggerElementToParent();
+    }
+  }
+
+  /** TODO: internal */
+  ngOnDestroy() {
+    // Remove event listeners on the trigger element.
+    this._rippleManager.clearTriggerElement();
+  }
+
+  /** TODO: internal */
+  ngOnChanges(changes: { [propertyName: string]: SimpleChange }) {
+    // If the trigger element changed (or is being initially set), add event listeners to it.
+    const changedInputs = Object.keys(changes);
+    if (changedInputs.indexOf('trigger') !== -1) {
+      this._rippleManager.setTriggerElement(this.trigger);
+    }
+  }
+
+  /**
+   * Responds to the start of a ripple animation trigger by fading the background in.
+   */
+  start() {
+    this._rippleManager.fadeInRippleBackground(this.backgroundColor);
+  }
+
+  /**
+   * Responds to the end of a ripple animation trigger by fading the background out, and creating a
+   * foreground ripple that expands from the event location (or from the center of the element if
+   * the "centered" property is set or forceCenter is true).
+   */
+  end(left: number, top: number, forceCenter = true) {
+    this._rippleManager.createForegroundRipple(
```

Comment:
Use of 'window.requestAnimationFrame()' like tab component when update ink-bar?
@jelbourn @kara